src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentDesignTimeNodeWriter.cs

```diff
@@ -366,7 +366,11 @@ public override void WriteComponent(CodeRenderingContext context, ComponentInter
         CodeWriterExtensions.CSharpCodeWritingScope? typeInferenceCaptureScope = null;
         string typeInferenceLocalName = null;
 
-        if (node.TypeInferenceNode == null)
+        var suppressTypeInference = ShouldSuppressTypeInferenceCall(node);
+        if (suppressTypeInference)
+        {
+        }
+        else if (node.TypeInferenceNode == null)
```

Comment:
```suggestion
        if (!suppressTypeInference && node.TypeInferenceNode == null)
```

Reply:
That's not the same? When `suppressTypeInference` is `true`, I want the whole original `if/else` block to do nothing. Your suggestion would skip over the first `if` branch but would instead take the `else` branch below it.

I could do the following instead if you prefer, but seems like unnecessary indentation:

```cs
if (!suppressTypeInference)
{
    if (node.TypeInferenceNode == null)
    {
        // ...existing code...
    }
    else
    {
        // ...existing code...
    }
}
```

Or this but that's seems less clear and more repetitive:

```cs
if (!suppressTypeInference && node.TypeInferenceNode == null)
{
    // ...existing code...
}
else if (!suppressTypeInference)
{
    // ...existing code...
}
```

Reply:
Ahh, I missed the following else block. Makes sense, thanks. I actually do prefer the `if(!suppressTypeInference)` version, but its purely preference so no need to change it.